docs/markdown/Using Pants/environments.md

```diff
@@ -0,0 +1,221 @@
+---
+title: "Environments: Cross-Platform or Remote Builds"
+slug: "environments"
+hidden: false
+createdAt: "2022-10-03T21:39:51.235Z"
+updatedAt: "2022-10-03T21:39:51.235Z"
+---
+Environments
+============
+
+By default, Pants will execute all sandboxed build work directly on localhost. But defining and using additional "environments" for particular targets allows Pants to transparently execute some or all of your build either:
+1. locally in Docker containers
+2. remotely via [remote execution](doc:remote-execution)
+3. locally, but with a non-default set of environment variables and settings (such as for cross-building)
+
+## Defining environments
+
+Environments are defined using environment targets:
+
+* [`local_environment`](doc:reference-local_environment) - Runs without containerization on localhost (which is also the default if no environment targets are defined).
+* [`docker_environment`](doc:reference-docker_environment) - Runs in a cached container using the specified Docker image.
```

Comment:
Does the `cached` here imply you have to `pull` any images outside of Pants before trying to use them?

Reply:
Pants automatically attempts to pull images. "Cached" here means that Pants will spin up a single container for that image and then run multiple processes in that container for build actions (via essentially `docker exec`).

Reply:
It might be worth specifying that "Docker" means "using the Docker installation on this machine"